master/docs/relnotes/index.rst

```diff
@@ -34,6 +34,10 @@ Deprecations, Removals, and Non-Compatible Changes
 Changes for Developers
 ~~~~~~~~~~~~~~~~~~~~~~
 
+* :class:`~buildbot.steps.source.base.Source` now has a ``doWithRetry`` method to help make it easier
```

Comment:
Yeh, that's a cool idea .. there's not much source-specific about it. Alternatives:
 - mixin
 - member variable (eg, self.retry = RetryHelper(retryParams)); this would let you have multiple "retry regimes"
 - move to Shell/BuildStep without utility.

What do you think?